xtable-core/src/main/java/org/apache/xtable/hudi/ConfigurationBasedPartitionSpecExtractor.java

```diff
@@ -29,20 +29,21 @@
 import org.apache.xtable.model.schema.InternalPartitionField;
 import org.apache.xtable.model.schema.InternalSchema;
 import org.apache.xtable.schema.SchemaFieldFinder;
+import org.apache.xtable.model.schema.PartitionFieldSpec;
 
 /**
  * Parses the InternalPartitionFields from a configured list of specs with the format
  * path:type:format for date types or path:type for value types.
  */
 @AllArgsConstructor
-public class ConfigurationBasedPartitionSpecExtractor implements HudiSourcePartitionSpecExtractor {
+public class ConfigurationBasedPartitionSpecExtractor implements PathBasedPartitionSpecExtractor {
   private final HudiSourceConfig config;
```

Comment:
Can we update this to take in the `List<PartitionFieldSpec>` directly so we can reuse it for the parquet source as well?

Reply:
Thanks! That is what I was looking for